docs/src/content/concepts-certificates.md

```diff
@@ -198,16 +195,59 @@ openssl req -x509 -new -nodes -key ca.key -sha256 -out ca.crt -addext keyUsage=c
 cat ca.key ca.crt > mitmproxy-ca.pem
 ```
 
-## Using a client side certificate
+## Mutual TLS and client certificates
+
+Mutual TLS (mTLS) typically is used for the server to verify the client's
+identity, in addition to the more common validation of server identity (is the
+server the one they claim they are).
+
+If the server wants to verify the clients identity using mTLS, the server sends
+a `CertificateRequest` message to the client as part of the TLS handshake. The
+client then needs to provide its cert and proof ownership of the private key by
+generating a signature.
+
+Client certificates, just as the more commonly known server certificates, are
+full x509 certs with a public and private key, commonly stored as PEM
+certificate file.
+
+Some internet protocols use client and server certificates as a form of
+authentication and authorization, e.g., the MQTT protocol used in
+Internet-of-Things networks. To intercept such protocols using mitmproxy, both
+the client needs to asked to establish mTLS, and the upstream server needs to
+believe that mitmproxy is a mTLS-compliant device. Such protocols require both
+options outlined in the next sections for mitmproxy to work correctly.
```

Comment:
Why are both options required? I guess there may be some applications that do check this, but generally speaking (especially when using OpenSSL APIs) most clients will work just fine without `request_client_cert`. The important bit is that mitmproxy does have the client cert to authenticate upstream, but `request_client_cert` is typically optional. Do you have clients where this is a hard requirement?

Reply:
Yes, this is a (at least in my visibility) common requirement of IoT devices and related firmware connectivity libraries. I'll rephrase this docs section to point out that this is not a hard requirement for every protocol and device.